apps/meteor/tests/end-to-end/api/07-incoming-integrations.js

```diff
@@ -165,6 +175,30 @@ describe('[Incoming Integrations]', function () {
 				.expect(200)
 				.end(done);
 		});
+
+		it('should not send a message for a channel that is not in the webhooks configuration', (done) => {
+			request
+				.post(`/hooks/${integration._id}/${integration.token}`)
+				.send({
+					text: 'Example message',
+					channel: [testChannelName],
+				})
+				.expect(200);
+
+			request
+				.get(api('channels.messages'))
+				.set(credentials)
+				.query({
+					roomId: channel._id,
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', true);
+					expect(res.body).to.have.property('messages', []);
```

Comment:
```suggestion
					expect(res.body).to.have.property('messages').and.to.be.an('array');
					expect(res.body.messages).to.eql([]);
```
You need deep equality here, this should work

Reply:
updated!